src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DEFAULT_INDEX = 0;
+    private static final int DIVIDER = 2;
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> current = new Node<>(value);
+        if (first == null) {
+            first = current;
+        } else {
+            current.prevElement = last;
+            last.nextElement = current;
+        }
+        last = current;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckerForAdding(index);
+        if (size() == DEFAULT_INDEX) {
+            add(value);
+            return;
+        }
+        if (index == DEFAULT_INDEX) {
+            Node<T> current = first;
+            first = new Node<>(value);
+            first.nextElement = current;
+            current.prevElement = first;
+            size++;
+            return;
+        }
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        Node<T> current = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(value);
+        newNode.prevElement = current.prevElement;
+        newNode.nextElement = current;
+        current.prevElement.nextElement = newNode;
+        current.prevElement = newNode;
+        size++;
     }
```

Comment:
I will not request changes if you leave it as it is

and at lines 44-50 code is more readable because you wrote `Node<T> newNode = new Node<>(value);` 
and then it is clear what we do when we write `newNode.prevElement = current.prevElement;`

but at 33-36 lines, you wrote `first = new Node<>(value);` 
and when we write `first.nextElement = current;` it is harder to read and understand what is going on